### DISCUSSION TITLE
ecChronos in VMs

### DISCUSSION THREAD

**Discussion by Vagg-eng** [Ericsson/ecchronos#783] [General]
Hello,
We tried to integrate ecChronos in VMs, not in containerized deployment.
Some questions about the product and the implementation to 
provide us some  insights.
1) Do you know if ecChronos has been deployed in VMs?

2) We've noticed that there are 2 alarms (warning and error) when something goes wrong 
   with the repairs. It is not clear how these are sent out. It gives me the 
   impression that ecChronos generates metrics which are scraped by PM server and the 
   Alarms are generated there according to defined rules about the metrics. Is this the 
   case? 
   If yes which are the metrics which are related to Alarms.
   In VMs we use SNMP agent and Manager which sends and receives Alarms. Iis there any 
   possibility your implementation to cover this use case?
   Also are the Alarms  logged to a log file?
   
   
3) One last question is how to enable debug log. I have contents in log file but not in debug log.
   I tried changing  root level to DEBUG in conf/logback.xml but no luck
   
BRs,
   Vaggelis

**Comment by tommystendahl** [comment]
Hi,
Thanks for reaching out.

1. Yes, it can be deployed in VMs.
2. The default alarm implementation loggs the alarms in the log file. There is no implementation for SNMP that can be done by implementing the interface RepairFaultReporter in the fm module.
3. To enable the debug log add this section
```
<root level="DEBUG">
    <appender-ref ref="STDOUT" />
    <appender-ref ref="FILE_DEBUG" />
</root>
```
In the conf/logback.xml and restart ecChronos. See docs/TROUBLESHOOTING.md

Regards,
Tommy
